source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,378 @@
+        maximum_refinement_level = Utilities::string_to_int(prm.get("Initial global refinement")) + Utilities::string_to_int(prm.get("Initial adaptive refinement"));
+        prm.enter_subsection("Isotherms");
+        {
+          exclude_composition = Utilities::string_to_int(prm.get("Exclude composition"));
+          std::vector<std::string> isotherms_outer_loop = Utilities::split_string_list(prm.get ("List of isotherms"),';');
+          // process the List of isotherms to get all data out of the structure
+          isotherms.resize(isotherms_outer_loop.size(),std::vector<std::string>(4,"0"));
+          // loop through all given isotherms
+          for ( unsigned int isotherm_line = 0; isotherm_line != isotherms.size(); ++isotherm_line )
+            {
+              std::vector<std::string> isotherms_inner_loop = Utilities::split_string_list(isotherms_outer_loop[isotherm_line]);
+              // The minimum and maximum refinemnt levels respecively
+              isotherms[isotherm_line][0]=isotherms_inner_loop[0];
+              isotherms[isotherm_line][1]=isotherms_inner_loop[1];
+              // The temperatures for the min and max isotherms respectively
+              isotherms[isotherm_line][2]=isotherms_inner_loop[2];
+              isotherms[isotherm_line][3]=isotherms_inner_loop[3];
+
+            }
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MeshRefinement
+  {
+    ASPECT_REGISTER_MESH_REFINEMENT_CRITERION(Isotherms,
+                                              "isotherms",
+                                              "A mesh refinement criterion that ensures a "
+                                              "maximum and minimum refinement level between "
+                                              "two temperatures (isotherms), with the possibility "
+                                              "to exclude a composition from this criterion. "
+                                              "To accomplish this there are two parameters "
+                                              "available: 'Exclude composition' and 'List of "
+                                              "isotherms'. The first parameter takes the "
+                                              "number of the compositional field and excludes "
+                                              "it from this plugin. "
+                                              "The second parameters takes a list of isotherm "
+                                              "parameters for one isotherm separated by a ';'. "
+                                              "Each line of isotherm parameters contains four "
+                                              "subparameters. The first subparameter sets "
+                                              "the minimun refinement level for this isotherm, "
```

Comment:
-> minimum

Reply:
I also don't quite understand what "minimum refinement level for this isotherm" means. An isotherm is a two-dimensional surface in a three-dimensional model. It is not a cell. Do you mean "minimum refinement level for all cells whose temperature (at at least one quadrature points) lies between the value of this and the following isotherm temperature" or similar?